src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -59,8 +60,12 @@ internal abstract partial class AbstractInitializeMemberFromParameterCodeRefacto
             // to an existing matching field/prop if we can find one, or add a new field/prop
             // if we can't.
 
+            var rules = await document.GetNamingRulesAsync(FallbackNamingRules.RefactoringMatchLookupRules, cancellationToken).ConfigureAwait(false);
+            var parameterNameMinusPrefixAndSuffix = NamingStyle.GetBaseName(parameter, rules);
+            var parameterWordParts = GetParameterWordParts(parameterNameMinusPrefixAndSuffix);
```

Comment:
i feel like it might be worthwhile to combine all of tihs into a helper struct.

Reply:
that way this sort of common operation could be easily reapplied to other features taht might need it.